src/main/java/com/hbm/world/gen/component/Component.java

```diff
@@ -663,7 +673,44 @@ protected void fillWithBlocks(World world, StructureBoundingBox box, int minX, i
 			}
 		}
 	}
-	
+	/** Have you ever wanted Glyphids in your building? No? Too bad! **/
+	protected void infest(World world, StructureBoundingBox box, int minX, int minY, int minZ, int maxX, int maxY, int maxZ, int maxSize, Random rand){
+		if(!enableInfestation)
+			return;
+
+		PollutionHandler.PollutionPerWorld ppw = PollutionHandler.perWorld.get(world);
+
+		int pollutedLocations = ppw == null ? 0 : ppw.pollution.size();
+		int infestChance = Math.max((int)MobConfig.baseInfestChance - (pollutedLocations/10), 1);
+
+		if(rand.nextInt(infestChance + 1) != 0 || getYWithOffset(minY) < box.minY || getYWithOffset(maxY) > box.maxY)
+			return;
+
+		ArrayList<Vec3> hives = new ArrayList<>();
+
+		for(int x = minX; x <= maxX; x++) {
+
+			for(int z = minZ; z <= maxZ; z++) {
+				int posX = getXWithOffset(x, z);
+				int posZ = getZWithOffset(x, z);
+
+				if(posX >= box.minX && posX <= box.maxX && posZ >= box.minZ && posZ <= box.maxZ) {
+
+					for(int y = minY; y <= maxY; y++) {
+						int posY = getYWithOffset(y);
+
+						if (rand.nextInt(1000) <= 8 + (pollutedLocations/10) - maxSize * 2 && hives.size() < maxSize) {
+                            if(hives.size() < 3 || hives.get(hives.size() - 1).squareDistanceTo(hives.get(hives.size() - 2)) <= 100) {
+								GlyphidHive.generateBigOrb(world, posX, posY, posZ, world.rand);
```

Comment:
This seems like a bad idea, all these structures use the more elegant component system which does checks for bounding boxes and prevents things like cascading chunk lag, and you've bashed the old "stupid" system into it which just places blocks in an area.